pages/collaborators.tsx

```diff
@@ -0,0 +1,230 @@
+import { useState, useMemo, useRef, useEffect } from "react"
+import { NextSeo } from "next-seo"
+import { UserLists } from "../components/UserLists"
+import { useUserId } from "../lib/web/utils"
+import { Tournaments } from "../components/Tournaments"
+import { api } from "../lib/web/trpc"
+import { TeamListContent } from "../components/TeamListContent"
+import { TournamentContent } from "../components/TournamentContent"
+import { signInToFatebook } from "../lib/web/utils"
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronDoubleDownIcon,
+  ChevronDoubleUpIcon,
+} from "@heroicons/react/20/solid"
+import {
+  ChevronLeftIcon as ChevronLeftIconLarge,
+  ChevronRightIcon as ChevronRightIconLarge,
+} from "@heroicons/react/24/solid"
+
+export default function CollaboratorsPage() {
+  const userId = useUserId()
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [isOverlayVisible, setIsOverlayVisible] = useState(false)
+  const [overlayTop, setOverlayTop] = useState("4rem")
+
+  const tournamentsQ = api.tournament.getAll.useQuery({})
+  const userListsQ = api.userList.getUserLists.useQuery()
+
+  type CarouselItemType =
+    | { type: "userList"; id: string; name: string; data: any }
+    | { type: "tournament"; id: string; name: string; data: any }
+
+  const items: CarouselItemType[] = useMemo(() => {
```

Comment:
nit: a couple things:

1. no need to memoise this I think, `map` is fast (possible I'm wrong about this)
2. I generally prefer not defining types like this and letting the compiler and IDE infer them
3. could then be written more concisely I think, e.g. something like

```
const items = [
  ...(userListsQ.data?.map((userList) => ({
    type: "userList" as const,
    id: userList.id,
    name: userList.name,
    data: userList,
  })) || []),
  ...(tournamentsQ.data?.map((tournament) => ({
    type: "tournament" as const,
    id: tournament.id,
    name: tournament.name,
    data: tournament,
  })) || [])
]
```

Reply:
Part of me thinks having the compiler infer types defeats the point of using TypeScript in the first place, but I see your point too, I've changed it

Reply:
I think best thing about Typescript because it can catch errors without having to run the code (e.g. I change a function definition and it tells me where all the errors are), I think inferring types generally is good for that (pretty lightly held, I have not looked at the philosophy on this)